app/controllers/proposals_controller.rb

```diff
@@ -170,4 +172,17 @@ def set_careers
     @careers = Person.where(id: @proposal.participants.pluck(:person_id))
                      .pluck(:academic_status)
   end
+
+  def check_file
+    return if File.exist?("#{Rails.root}/tmp/#{latex_temp_file}")
+
+    if params[:action] == "latex_field"
+      input = latex_params[:latex]
+      @data = ProposalPdfService.new(@proposal.id, latex_temp_file, input)
+                                .generate_latex_file
+    end
+    File.new("#{Rails.root}/tmp/#{latex_temp_file}", 'w') do |io|
+      io.write(@data)
+    end
```

Comment:
`ProposalPdfService. generate_latex_file` already writes latex_temp_file to file